packages/agent/src/step/Step.ts

```diff
@@ -38,12 +38,21 @@ export abstract class Step {
     this.state = { type: "running" };
     this.run.onStepExecutionStarted({ step: this });
 
-    const result = await this._execute();
-    this.state = result;
-
-    this.run.onStepExecutionFinished({ step: this, result });
-
-    return result;
+    try {
+      const result = await this._execute();
+      this.state = result;
+      this.run.onStepExecutionFinished({ step: this, result });
```

Comment:
`onStepExecutionFinished` should also be called in case of an error.

Reply:
done: https://github.com/lgrammel/js-agent/compare/cd7f250f16f97ddc7bbc56db3b7d12578376241f..804b8012f35b5ab6859d73d41d33c400940ecfa1